src/core/interaction-creator.ts

```diff
@@ -0,0 +1,194 @@
+import * as ts from 'typescript';
+import * as tsMorph from 'ts-morph';
+import {mapJsDocsIntoInteraction} from './js-docs-into-interaction';
+import {getBasicRepresentationOfType, getReturnTypeOfFunction} from './typescript-types';
+import {changeObjectRepresentationIntoExample, changeObjectRepresentationIntoMatchingRules} from './pact-generating';
+import {isEmptyObject} from '../utils/object-type';
+import qs from 'qs';
+import {Provider} from './read-pacts-config';
+import {getDefaultResponseStatusForInteraction} from './default-response-status';
+
+export interface Interaction {
+    description?: string;
+    request: {
+        method?: string;
+        path?: string;
+        body?: unknown;
+        query?: string;
+        matchingRules?: object;
+        headers?: Record<string, string>;
+    };
+    response: {
+        status?: number;
+        body?: unknown;
+        matchingRules?: object;
+        headers?: Record<string, string>;
+    };
+}
+
+export class InteractionCreator {
+    constructor(private readonly sourceFile: tsMorph.Node, private readonly provider: Provider) {}
+
+    public static getAllInteractionsInFile(sourceFile: tsMorph.Node, provider: Provider) {
+        return new InteractionCreator(sourceFile, provider).findAllInteractions();
+    }
+
+    public findAllInteractions() {
+        const allJsDocsNodes = this.sourceFile.getDescendantsOfKind(ts.SyntaxKind.JSDocComment);
+        const allPactJsDocsNodes = allJsDocsNodes.filter((jsDocNode) =>
+            jsDocNode
+                .getChildrenOfKind(ts.SyntaxKind.JSDocTag)
+                .find((jsDocTag) => jsDocTag.getFirstChildByKind(ts.SyntaxKind.Identifier)?.getText() === 'pact'),
+        );
+        return allPactJsDocsNodes.map(this.getInteractionForPactJsDoc, this);
+    }
+
+    private getInteractionForPactJsDoc(pactJsDoc: tsMorph.Node): Interaction {
```

Comment:
This method should rather get the parent node as parameter (the `nodeWithUsagePactJsDoc`), and only operate on its children.

Reply:
I disagree.
the name of that function points that it is taken from pact js doc node.
``nodeWithUsagePactJsDoc`` is not JsDoc, it is some AST node which wrapes our JsDoc and in most cases it is a function node.